WordPress/gutenberg
# Plugin: require minimum WordPress 6.2

## What?

WordPress 6.3 has been released: https://wordpress.org/news/2023/08/lionel/ 🎹 

This PR removes 6.2 compatibility code, thereby making WordPress 6.2 the minimum version required for the Gutenberg plugin.

This is a first pass. There might be 6.2 code and other compat work to be done after this, for example, ensuring that any deprecated functions are replaced, duplicated unit tests removed.

## How?
- Removing the `compat/6.2` folder
- Replacing 6.2 `gutenberg_` methods with their core equivalents
- Removed some tests already covered in core since 6.2

## Testing Instructions
Ensure things are not broken.
Run the Gutenberg plugin with WP 6.2, and confirm editors work as before.
Run the Gutenberg plugin with WP 6.3, and confirm editors work as before.

If you're using `wp-env` you can add an `.wp-env.override.json` file to Gutenberg root:

```json
{
	"core": "WordPress/WordPress#6.2"
}
```

or

```json
{
	"core": "WordPress/WordPress#6.3"
}
```

and run `npm run wp-env start --update`
